homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -0,0 +1,235 @@
+"""The Azure Data Explorer integration."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from datetime import datetime
+import json
+import logging
+from typing import Any
+
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import MATCH_ALL
+from homeassistant.core import Event, HomeAssistant, State
+from homeassistant.exceptions import ConfigEntryNotReady, IntegrationError
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.json import JSONEncoder
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util.dt import utcnow
+
+from .client import AzureDataExplorerClient
+from .const import (
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DATA_FILTER,
+    DATA_HUB,
+    DEFAULT_MAX_DELAY,
+    DOMAIN,
+    FILTER_STATES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+            },
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
+    """Activate ADX component from yaml.
+
+    Adds an empty filter to hass data.
+    Tries to get a filter from yaml, if present set to hass data.
+    If config is empty after getting the filter, return, otherwise emit
+    deprecated warning and pass the rest to the config flow.
+    """
+
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    if DOMAIN not in yaml_config:
+        return True
+    hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN][CONF_FILTER]
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Do the setup based on the config entry and the filter from yaml."""
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    adx = AzureDataExplorer(hass, entry)
+    try:
+        await adx.test_connection()
+    except KustoServiceError as exp:
+        _LOGGER.error(exp)
+        raise IntegrationError(
+            "Could not find Azure Data Explorer database or table"
+        ) from exp
+    except KustoAuthenticationError as exp:
+        _LOGGER.error(exp)
+        raise ConfigEntryNotReady(
```

Comment:
```suggestion
        raise ConfigEntryAuthFailed(
```

Reply:
by changing this I get this Error: Handler ADXConfigFlow doesn't support step reauth

Reply:
You will need to implement reauth support in your Config Flow, see https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#reauthentication. `ConfigEntryNotReady` will keep retrying, where `ConfigEntryAuthFailed` will raise an error and ask the user to reauthenticate.

Reply:
I want to add that we usually like to have the reauth flow in a follow up PR. So I think this is advice isn't relevant now but will be the end goal. For now, to stop retrying you can do:
```suggestion
        return False
```